File: stack.go
package main

import "fmt"

type Stack []int

func (s *Stack) Push(val int) {
	*s = append(*s, val)
}

func (s *Stack) Pop() int {
	if s.IsEmpty() {
		fmt.Println("Cannot pop, stack is empty.")
		return -1
	}
	index := len(*s) - 1
	element := (*s)[index]
	*s = (*s)[:index]
	return element
}

func (s *Stack) IsEmpty() bool {
	return len(*s) <= 0
}

func (s *Stack) Print() {
	for _, element := range *s {
		fmt.Println(element)
	}
}

func main() {
	s := Stack{}
	s.Push(3)
	s.Push(4)
	s.Push(5)
	s.Push(6)
	s.Print()
	fmt.Println(s.Pop())
	fmt.Println(s.Pop())
	fmt.Println(s.Pop())
	fmt.Println(s.Pop())
	fmt.Println(s.Pop())

}

File: stack.py

# without pop push methods i guess

class Stack():
    def __init__(self):
        self.data = []

    def show(self):
        for i in self.data:
            print(i)

    def push(self, val):
        self.data.append(val)

    def pop(self):
        if len(self.data) <= 0:
            print("cannot pop, empty stack")
            return
        v = self.data[len(self.data)-1]
        self.data = self.data[:-1]
        return v


s = Stack()
s.push("hello")
s.push("world")
s.push("this")
s.push("is")
s.push("a")
s.push("stack")

s.show()

print()
print(s.pop())
print(s.pop())
print(s.pop())
print(s.pop())
#print(s.pop())
#print(s.pop())

print()
s.show()